gitlab/data_source_gitlab_users.go

```diff
@@ -0,0 +1,245 @@
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"users": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"username": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"email": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"is_admin": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"can_create_group": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"can_create_project": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"projects_limit": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"created_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"state": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"external": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"extern_uid": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"organization": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"two_factor_enabled": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceGitlabUsersRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+
+	listUsersOptions, id, err := expandGitlabUsersOptions(d)
+	if err != nil {
+		return err
+	}
+	users, _, err := client.Users.ListUsers(listUsersOptions)
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("users", flattenGitlabUsers(users))
+	d.SetId(fmt.Sprintf("%d", id))
+
+	return nil
+}
+
+func flattenGitlabUsers(users []*gitlab.User) []interface{} {
+	usersList := []interface{}{}
+
+	for _, user := range users {
+		values := map[string]interface{}{
+			"id":                 user.ID,
+			"username":           user.Username,
+			"email":              user.Email,
+			"name":               user.Name,
+			"is_admin":           user.IsAdmin,
+			"can_create_group":   user.CanCreateGroup,
+			"can_create_project": user.CanCreateProject,
+			"projects_limit":     user.ProjectsLimit,
+			"state":              user.State,
+			"external":           user.External,
+			"extern_uid":         user.ExternUID,
+			"organization":       user.Organization,
+			"two_factor_enabled": user.TwoFactorEnabled,
+		}
+
+		if user.CreatedAt != nil {
+			values["created_at"] = user.CreatedAt.String()
+		}
+
+		usersList = append(usersList, values)
+	}
+
+	return usersList
+}
+
+func expandGitlabUsersOptions(d *schema.ResourceData) (*gitlab.ListUsersOptions, int, error) {
+	listUsersOptions := &gitlab.ListUsersOptions{}
+	optionsHash := ""
```

Comment:
Maybe use [`strings.Builder()`](https://golang.org/pkg/strings/#Builder.String) now that the provider is [targeting golang `1.11.x`](https://github.com/terraform-providers/terraform-provider-gitlab/commit/81a53e359ed80e0900f6178c91a39f07130b9ef3#diff-354f30a63fb0907d4ad57269548329e3)?

Reply:
Done.